scripts/docs-lint/ascii-diagram-detector.sh

```
@@ -0,0 +1,58 @@
+# Common ASCII fallbacks like `+--+` and `|  |` in tables are NOT flagged —
+# they're legitimate Markdown tables.
+#
+# Exits 1 if any box-drawing characters are found in docs/architecture/.
+
+set -euo pipefail
+
+ROOT="$(git rev-parse --show-toplevel)"
+cd "$ROOT"
+
+# Unicode box-drawing block: ─ │ ┌ ┐ └ ┘ ├ ┤ ┬ ┴ ┼ ═ ║ ╔ ╗ ╚ ╝ ╠ ╣ ╦ ╩ ╬ etc.
+pattern=$'[─-╿]'
```

Shellcheck (0.11.0)

[warning] pattern appears unused. Verify use (or export if used externally).

(SC2034)